Models/Listing.cs

```diff
@@ -1,24 +1,34 @@
-﻿using System;
+﻿using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using System;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 
 namespace Karma.Models
-{ 
+{
 	public class Listing
 	{
-		public string Id { get; set; }
+        public string? Id { get; set; }
 
+		[Required]
+		[StringLength(20)]
 		public string Name { get; set; }
 
+		[StringLength(200)]
 		public string Description { get; set; }
 
+		[Required]
+		[Range(1, 100)]
```

Comment:
Naked constants seem like a bad practice.
Some solutions (I can think of): readonly static field in the class like maxNameLength, or another(more global) class with constants.